File: Chapter 2-5-UINavigationBar Tint Not Working/Swift 2.0/OldSchoolCArray/OldSchoolCArray/ViewController.m
//
//  ViewController.m
//  OldSchoolCArray
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
  
  int count = 5;
  
  NSMutableArray *anotherArray = [[NSMutableArray alloc] init];
  
  
  // cArray is an array of “id” elements
  // count is the number of entries the cArray will have
  
  id cArray [count];
  id __strong *arrayPtr = cArray;
  
  /* populate the array
   arrayPtr[0] = …
   arrayPtr[1] = …
   arrayPtr[2] = …
   etc…
   */
  
  [anotherArray
   enumerateObjectsWithOptions:NSEnumerationConcurrent
   usingBlock:^(id obj, NSUInteger idx, 
                BOOL *stop) {
     arrayPtr[idx] = newObject;
   }];
  
  
  
}

- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}

@end

File: Chapter 2-8-Magical Way of Counting Elements/Swift 2.0/TYPE 2/CountingElements/CountingElements/ViewController.swift
//
//  ViewController.swift
//  CountingElements
//
//  Copyright © 2015 - by Magno Urbano - All rights reserved.
//  Part of the book Troubleshooting Xcode, published by Apress
//

import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do any additional setup after loading the view, typically from a nib.
    
    // creating the dictionaries
    let dict1 = ["object" : "television", "selected": true]
    let dict2 = ["object" : "phone",      "selected": false]
    let dict3 = ["object" : "book",       "selected": true]
    
    // creating the array
    let array = [dict1, dict2, dict3]
    
    // the magic command
    let filteredArray = array.filter({$0["selected"] == true})
    
    let result = filteredArray.count
    print(result) // this will print 2 on console

  }

  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }


}

